File: 프로그래머스/lv1/12901. 2016년/2016년.py
from functools import reduce
def solution(a, b):
    DAYS = {0:'THU', 1: 'FRI', 2:'SAT', 3:'SUN', 4:'MON', 5:'TUE', 6:'WED'}
    dates = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    targetMonths = dates[0:a-1]
    totalDays = reduce(lambda acc, cur: acc+cur,targetMonths, b);
    return DAYS[totalDays%7]

File: 프로그래머스/lv1/12901. 2016년/2016년.js
function solution(a, b) {
    const DAYS = {0:'THU', 1: 'FRI', 2:'SAT', 3:'SUN', 4:'MON', 5:'TUE', 6:'WED'};
    const dates = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    const targetMonths = dates.slice(0, a-1);
    const totalDays = targetMonths.reduce((acc, cur) => acc+cur,b);
    return DAYS[totalDays%7];
}